src/Runtime/Distributions/Automata/Automaton.Condensation.cs

```diff
@@ -108,7 +108,7 @@ public class Condensation
             /// </param>
             internal Condensation(State root, Func<Transition, bool> transitionFilter, bool useApproximateClosure)
             {
-                Debug.Assert(root != null, "A valid root node must be provided.");
+                Debug.Assert(!root.IsNull, "A valid root node must be provided.");
```

Comment:
How does this work? I don't see `State` having changed to a `struct`. If it was a `struct` to begin with, `root != null` would make no sense, if it's a class, the `IsNull` property may not exist is if `root` is `null`.

Reply:
nevermind, I found the missing diff